### PR TITLE
Adding gulp as a prerequisite

### PR DESCRIPTION
**Pull request by hestellez**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

File: docs/how-to-setup-freecodecamp-locally.md [hunk 1]
```diff
@@ -116,6 +116,7 @@ Start by installing these prerequisite software.
 | [MongoDB Community Server](https://docs.mongodb.com/manual/administration/install-community/) | `3.6`   | [Release Notes](https://docs.mongodb.com/manual/release-notes/), Note: We currently on `3.6`, an [upgrade is planned](https://github.com/freeCodeCamp/freeCodeCamp/issues/18275).
 | [Node.js](http://nodejs.org)                | `8.x`   | [LTS Schedule](https://github.com/nodejs/Release#release-schedule) |
 | npm (comes bundled with Node)               | `6.x`   | Does not have LTS releases, we use the version bundled with Node LTS |
+| gulp                                        | `4.0`   | It's called in the [curriculum scripts](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/package.json) |
```
Comment (RandellDawson) [inline docs/how-to-setup-freecodecamp-locally.md]: @raisedadead Is this necessary?

**Comment by RandellDawson** [comment]
@hestellez You checked the box stating:
> * All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
However, your PR contains both Spanish and English file changes.  Can you please remove the Spanish file from this PR and move it to a separate PR?

**Comment by hestellez** [comment]
@RandellDawson I made a pull request for change:
https://github.com/freeCodeCamp/freeCodeCamp/pull/35152 Gulp as a prerequisite.
https://github.com/freeCodeCamp/freeCodeCamp/pull/35153 Indentation spaces to example code.

**Comment by RandellDawson** [comment]
@hestellez You only needed to revert the changes for the Spanish file (`curriculum/challenges/spanish/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.spanish.md`) on this PR and leave the other 2 files on here.    

I have closed the Gulp one (35152), so now you just need to revert the changes to `curriculum/challenges/spanish/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.spanish.md` on this PR and push the changes back to this PR, so it will no longer have those changes.

Thank you.

**Pull request closed without merging**